enterprise/api_client/sso_orchestrator.py

```diff
@@ -0,0 +1,140 @@
+import base64
+import requests
+
+from http import HTTPStatus
+from urllib.parse import urljoin
+
+from rest_framework.reverse import reverse
+
+from django.conf import settings
+
+from enterprise.errors import ClientError
+from enterprise.utils import (
+    get_configuration_value,
+    get_sso_orchestrator_api_base_url,
+    get_sso_orchestrator_configure_path,
+    get_sso_orchestrator_basic_auth_username,
+    get_sso_orchestrator_basic_auth_password,
+)
+
+class EnterpriseSSOOrchestratorApiClient:
+    """
+    The enterprise API client to communicate with the SSO Orchestrator API. Reads conf settings values to determine
+    orchestration paths and credentials. 
+    
+    Required settings values:
+    - LMS_ROOT_URL
+    - SSO_ORCHESTRATOR_API_BASE_URL
+    - SSO_ORCHESTRATOR_CONFIGURE_PATH
+    - SSO_ORCHESTRATOR_BASIC_AUTH_USERNAME
+    - SSO_ORCHESTRATOR_BASIC_AUTH_PASSWORD
+    """
+
+    def __init__(self):
+        self.base_url = get_sso_orchestrator_api_base_url()
+        self.session = None
+
+    def _get_orchestrator_callback_url(self, config_pk):
+        """
+        get the callback url for the SSO Orchestrator API
+        """
+        lms_base_url = get_configuration_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
+        if not lms_base_url:
+            raise ClientError(
+                "Failed to get LMS_BASE_URL from configuration.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        path = reverse(
+            'enterprise-customer-sso-configuration-orchestration-complete',
+            kwargs={'configuration_uuid': config_pk},
+        )
+        return f"{lms_base_url}{path}"
+
+    def _get_orchestrator_configure_url(self):
+        """
+        get the configure url for the SSO Orchestrator API
+        """
+        # probably want config value validated for this
+        return urljoin(self.base_url, get_sso_orchestrator_configure_path()) 
+
+    def _create_auth_header(self):
+        """
+        create the basic auth header for requests to the SSO Orchestrator API 
+        """
+        orchestrator_username = get_sso_orchestrator_basic_auth_username()
+        if not orchestrator_username:
+            raise ClientError(
+                "Failed to create SSO Orchestrator auth headers: username required.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        orchestrator_password = get_sso_orchestrator_basic_auth_password()
+        if not orchestrator_password:
+            raise ClientError(
+                "Failed to create SSO Orchestrator auth headers: password required.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        return f"Basic {base64.b64encode(f'{orchestrator_username}:{orchestrator_password}'.encode('utf-8')).decode()}"
```

Comment:
doesn't `requests` handle this for us? https://docs.python-requests.org/en/latest/user/authentication/#basic-authentication

Reply:
it's very likely haha